shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -236,3 +238,182 @@ async fn test_hot_reload_with_zero_connections() {
+    let cert_dir_old = Path::new("/tmp/shotover-cert-test-old");
+    let cert_dir_new = Path::new("/tmp/shotover-cert-test-new");
+
+    // Clean up any existing certificate directories from previous test runs
+    let _ = std::fs::remove_dir_all(cert_dir_old);
+    let _ = std::fs::remove_dir_all(cert_dir_new);
+
+    // Generate old certificate set using test_helpers
+    test_helpers::cert::generate_test_certs(cert_dir_old);
+
+    // Generate new certificate set (distinct from old)
+    // The certificates will have different serial numbers and timestamps
+    test_helpers::cert::generate_test_certs(cert_dir_new);
+
+    let _compose = docker_compose("tests/test-configs/hotreload/docker-compose.yaml");
+
+    // Start the old shotover instance with old certificates
+    let shotover_old = shotover_process("tests/test-configs/hotreload/topology-tls-old.yaml")
+        .with_log_name("old_tls")
+        .with_hotreload_socket(socket_path)
+        .with_config("tests/test-configs/shotover-config/config_metrics_disabled.yaml")
+        .start()
+        .await;
+
+    // Create TLS clients configured with old CA certificate
+    let client_old =
+        create_tls_valkey_client_from_certs("127.0.0.1", 6380, "/tmp/shotover-cert-test-old");
+
+    // Establish multiple TLS connections to old instance
+    let mut connections = Vec::new();
+    for i in 0..10 {
+        let mut con = client_old.get_connection().unwrap();
+        // Store some data in each connection to verify persistence
+        let _: () = con
+            .set(format!("cert_key_{}", i), format!("cert_value_{}", i))
+            .unwrap();
+        connections.push(con);
+    }
+
+    // Set a counter to track across the certificate change
+    // Use "counter" key to match what assert_valkey_connection_works expects
+    let _: () = connections[0].set("counter", 0).unwrap();
+
+    // Verify all old connections work with old certificate
+    for (i, con) in connections.iter_mut().enumerate() {
+        let value: String = con.get(format!("cert_key_{}", i)).unwrap();
+        assert_eq!(value, format!("cert_value_{}", i));
+    }
+
+    // Start the new shotover instance with NEW certificates - triggers hot reload
+    let shotover_new = shotover_process("tests/test-configs/hotreload/topology-tls-new.yaml")
+        .with_log_name("new_tls")
+        .with_hotreload_socket(socket_path)
+        .with_config("tests/test-configs/shotover-config/config_metrics_disabled.yaml")
+        .start()
+        .await;
+
+    // Create a new TLS client configured with certificate
+    let client_new =
+        create_tls_valkey_client_from_certs("127.0.0.1", 6380, "/tmp/shotover-cert-test-new");
+    let mut con_new = client_new.get_connection().unwrap();
+
+    // Verify data persistence across certificate change
+    let value: String = con_new.get("cert_key_0").unwrap();
+    assert_eq!(value, "cert_value_0");
```

Comment:
This seems largely equivalent to `assert_valkey_connection_works`, should we use that instead?